provisioner/config.go

```diff
@@ -55,6 +55,9 @@ const (
 	// If it is not set then default value(90s) will be used
 	GraceTime        = "GraceTime"
 	DefaultGraceTime = 90
+
+	// FSGroupID defines the permissions of nfs share volume
+	FSGroupID = "FSGID"
```

Comment:
q_: Does Kubernetes pass the FSGUID related details via PVC? If it is not supported with an external provisioner - is this passed when we use CSI? 

One drawback from an operations point of view for this solution is: User has to create multiple storage classes if different applications need different FSGID values.

Reply:
> q_: Does Kubernetes pass the FSGUID related details via PVC? If it is not supported with an external provisioner - is this passed when we use CSI?

In PVC there is no placeholder for `FSGID`, it actually exists at pod level [pod.spec.securityContext.fsGroup](https://github.com/kubernetes/kubernetes/blob/b0010c2d9ea8f2d20ffca20da7846eb4bba6e456/pkg/apis/core/types.go#L2970-L2980). To support on PVC we can ask a user to specify `openebs.io/nfs-fsgid` annotation on PVC and during provisioning time provisioner will read from annotation and propagate it to NFS deployment.

Reply:
In CSI - there is a PodInfo object that is passed to volume request. Will it contain this information?

Reply:
No Kiran, Kubelet will not propagate any such information to the driver since kubelet itself takes responsibility to update volume group permissions

> In CSI - there is a PodInfo object that is passed to volume request. Will it contain this information?

It only contains sends information about pod metadata like {name, uid, namespace...}. We can't update ownership permissions of NFS volume by looking at a single pod since NFS volume can be consumed kind by different non-root applications.

FYI If the access mode is `ReadWriteMany` Kubelet will not update ownership permissions.